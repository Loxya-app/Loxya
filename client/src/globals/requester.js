import axios from 'axios';
import Config from '@/globals/config';
import Cookies from '@/utils/cookies';

const requester = axios.create({
    baseURL: Config.api.url,
    headers: Config.api.headers,
});

requester.interceptors.request.use(
    (_request) => {
        const { params, ...request } = _request;

        if (params) {
            Object.keys(params).forEach((name) => {
                if (params[name] === true) {
                    params[name] = '1';
                }
                if (params[name] === false) {
                    params[name] = '0';
                }
            });
            request.params = params;
        }

        const token = Cookies.get(Config.auth.cookie);
        if (token) {
            request.headers.Authorization = `Bearer ${token}`;
        }
        return request;
    },
    (error) => Promise.reject(error),
);

export default requester;
