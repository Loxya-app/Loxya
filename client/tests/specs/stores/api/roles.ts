import requester from '@/globals/requester';
import apiRoles from '@/stores/api/roles';
import data from '@fixtures/roles';

describe('Roles Api', () => {
    describe('all()', () => {
        it('parse the returned data correctly', async () => {
            jest.spyOn(requester, 'get').mockResolvedValue({ data: data.default() });
            expect(await apiRoles.all()).toMatchSnapshot();
        });
    });

    describe('create()', () => {
        it.each(data.default())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'post').mockResolvedValue({ data: datum });
            expect(await apiRoles.create({} as any)).toMatchSnapshot();
        });
    });

    describe('update()', () => {
        it.each(data.default())('parse the returned data correctly (with #$id)', async (datum: any) => {
            jest.spyOn(requester, 'put').mockResolvedValue({ data: datum });
            expect(await apiRoles.update(datum.id, {} as any)).toMatchSnapshot();
        });
    });
});
