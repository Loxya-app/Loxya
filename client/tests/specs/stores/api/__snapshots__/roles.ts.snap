// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Roles Api all() parse the returned data correctly 1`] = `
[
  {
    "id": 1,
    "is_used": true,
    "name": "Régisseur",
  },
  {
    "id": 2,
    "is_used": true,
    "name": "Technicien plateau",
  },
  {
    "id": 3,
    "is_used": true,
    "name": "Ingénieur du son",
  },
  {
    "id": 4,
    "is_used": true,
    "name": "Installateur",
  },
  {
    "id": 5,
    "is_used": false,
    "name": "Intervenant extérieur",
  },
]
`;

exports[`Roles Api create() parse the returned data correctly (with #1) 1`] = `
{
  "id": 1,
  "is_used": true,
  "name": "Régisseur",
}
`;

exports[`Roles Api create() parse the returned data correctly (with #2) 1`] = `
{
  "id": 2,
  "is_used": true,
  "name": "Technicien plateau",
}
`;

exports[`Roles Api create() parse the returned data correctly (with #3) 1`] = `
{
  "id": 3,
  "is_used": true,
  "name": "Ingénieur du son",
}
`;

exports[`Roles Api create() parse the returned data correctly (with #4) 1`] = `
{
  "id": 4,
  "is_used": true,
  "name": "Installateur",
}
`;

exports[`Roles Api create() parse the returned data correctly (with #5) 1`] = `
{
  "id": 5,
  "is_used": false,
  "name": "Intervenant extérieur",
}
`;

exports[`Roles Api update() parse the returned data correctly (with #1) 1`] = `
{
  "id": 1,
  "is_used": true,
  "name": "Régisseur",
}
`;

exports[`Roles Api update() parse the returned data correctly (with #2) 1`] = `
{
  "id": 2,
  "is_used": true,
  "name": "Technicien plateau",
}
`;

exports[`Roles Api update() parse the returned data correctly (with #3) 1`] = `
{
  "id": 3,
  "is_used": true,
  "name": "Ingénieur du son",
}
`;

exports[`Roles Api update() parse the returned data correctly (with #4) 1`] = `
{
  "id": 4,
  "is_used": true,
  "name": "Installateur",
}
`;

exports[`Roles Api update() parse the returned data correctly (with #5) 1`] = `
{
  "id": 5,
  "is_used": false,
  "name": "Intervenant extérieur",
}
`;
